backend/src/langchain.ts

```diff
@@ -73,51 +74,77 @@ async function getDocuments(filePath: string) {
 // join the configurable preprompt to the context template
 function getQAPromptTemplate(prePrompt: string) {
   if (!prePrompt) {
-    console.debug("Using default retrieval QA pre-prompt");
+    // use the default prePrompt
     prePrompt = retrievalQAPrePrompt;
   }
   const fullPrompt = prePrompt + qAcontextTemplate;
+  console.debug("QA prompt: " + fullPrompt);
   const template: PromptTemplate = PromptTemplate.fromTemplate(fullPrompt);
   return template;
 }
+// create and store the document vectors for each phase
+async function initDocumentVectors(openAiApiKey: string) {
+  if (!openAiApiKey) {
+    console.debug("No OpenAI API key set to initialise document vectors.");
+    return;
+  }
+  let docVectors: DocumentsVector[] = [];
+
+  for (const value of Object.values(PHASE_NAMES)) {
+    if (!isNaN(Number(value))) {
+      const phase = value as PHASE_NAMES;
+      // get the documents
+      const filePath: string = getFilepath(phase);
+      const documents: Document[] = await getDocuments(filePath);
+      // embed and store the splits
+      const embeddings = new OpenAIEmbeddings({
+        openAIApiKey: openAiApiKey,
+      });
+      const vectorStore: MemoryVectorStore =
+        await MemoryVectorStore.fromDocuments(documents, embeddings);
+      // store the document vectors for the phase
+      docVectors.push({
+        phase: phase,
+        docVector: vectorStore,
+      });
+    }
+  }
+  setVectorisedDocuments(docVectors);
```

Comment:
I'm wondering here, `vectorisedDocuments` isn't per user, but embeds a user's API key. Are we able to vectorise the docs  without embedding the key? And maybe embed it later when the user sets their key?